### PR TITLE
[improvement](nereids) Use a unified approach to deal with monotonic function in partition prune

### PR DESCRIPTION
**Pull request by feiniaofeiafei**
### What problem does this PR solve?
This pr use a unified approach to deal with monotonic function in partition prune. When adding a monotonic function in partition prune, we don't need to add visitXXX in OneRangePartitionEvaluator, only need to implement the Monotonic interface. 

### Release note

None

### Check List (For Author)

- Test 
    - [x] Regression test
    - [x] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run p0

**Review by morrySnow — DISMISSED** [review]

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/OneRangePartitionEvaluator.java [hunk 1]
```diff
@@ -820,15 +786,31 @@ private List<Map<Slot, PartitionSlotInput>> commonComputeOnePartitionInputs() {
...
         }
         int childIndex = func.getMonotonicFunctionChildIndex();
         Expression funcChild = func.child(childIndex);
+        Expression forNullable;
+        if (partitionSlotContainsNull.containsKey(funcChild)) {
+            forNullable = func.withConstantArgs(partitionSlotContainsNull.get(funcChild)
+                    ? new Nullable(funcChild) : new NonNullable(funcChild));
+        } else {
+            forNullable = func.withConstantArgs(new Nullable(funcChild));
+        }
```
Comment (924060929) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/OneRangePartitionEvaluator.java:811]:
```java
boolean isNullable = partitionSlotContainsNull.getOrDefault(funcChild, true);
Expression withNullable = func.withConstantArgs(
        isNullable ? new Nullable(funcChild) : new NonNullable(funcChild)
);
```

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by 924060929 — APPROVED** [review]

**Pull request merged**